### ISSUE TITLE
La commande \sameword ne parche pas dans l'environnement \pstart[\setRL]

### ISSUE THREAD

**Issue by yadaddi** [maieul/ledmac#835]
# Version
Package: reledmac 2019/12/09 v2.32.3 typesetting critical editions

# Regression

# Description

La commande \sameword ne produit pas le numéro de mot référencé dans la note critique en exposant, comme c'est le cas dans l'exemple en mode LR.

# Minimal Working Example

```latex
```
\documentclass[12pt]{article}
\usepackage[series={A},noend,noeledsec,noledgroup]{reledmac}
\usepackage{reledpar}
\usepackage{zref-abspage}
\usepackage{perpage}
\MakePerPage{footnote}
\usepackage{scrextend}
\usepackage{fontspec}
\usepackage{marginnote}
\usepackage{polyglossia}
\setmainlanguage{english}
\setotherlanguage[locale = tunisia]{arabic}
\setmainfont[Script = Arabic]{Amiri} 
\newfontfamily\arabicfont[Script = Arabic]{Amiri}
\widowpenalty=10000
\clubpenalty=10000
\raggedbottom
\Xnotefontsize[A]{\scriptsize}
\bhookgroupX[A]{\RTL}
\arrangementX[A]{paragraph}
\wrapcontentX[A]{\textarabic}
\Xarrangement{paragraph}
\Xbhookgroup[A]{\RTL}
\Xafterlemmaseparator[A]{0.5em}
\Xbeforelemmaseparator[A]{0.25em}
\Xlinerangeseparator{\null-\null}
\Xbeforeinserting{\LTR}

\makeatletter
\renewcommand{\Afootnoterule}{\right@footnoterule}
\makeatother
\firstlinenum{1}\linenumincrement{1} \lineation{page} 
\fnpos{familiar-critical} 
\usepackage{scrextend} 

\XeTeXinputnormalization 1 % for \sameword see reledmac, p. 28

\linenumberstyle{arabic}
\Xnotenumfont{\arabicfont\bfseries}

\begin{document}
\beginnumbering
\pstart
\linenummargin{right}
% Shows which word number in the critical footnote
Lupus \sameword{aut} canis \edtext{\sameword{aut}}{\Afootnote{et}} felix
\pend

\pstart[\setRL]
% does not show the word number in the critical footnote
Lupus \sameword{aut} canis \edtext{\sameword{aut}}{\Afootnote{et}} felix

\pend
\endnumbering
\end{document}
```
```

**Comment by maieul** [comment]
Je connais l'origine du bug, et je pense pouvoir le résoudre, mais sans
doute pas avant 2 semaine.

**Comment by yadaddi** [comment]
Merci! Ca me va très bien!

**Comment by maieul** [comment]
Je viens de trouver une solution de debug beaucoup plus simple. Merci de tester la branche issue835 et de me faire un retour.

**Comment by maieul** [comment]
désolé, il y a encore un bug. Je regarde.

**Comment by maieul** [comment]
le problème est bien plus compliqué que je ne le pensais. Et je ne vois pas de solution à court terme, et je n'ai pas el tems de m'y pencher.

La seule solution pour vous : passer en lualatex, qui gère correctement l'écriture dans les fichiers auxiliaires.

**Comment by yadaddi** [comment]
Merci d'avoir essayé! Je suis coincé dans un sense car j'ai rencontré un problème auparavant sur Reledmac-XeLaTeX-arabe. Robert Alessi m'avait conseillé d'utiliser LuaLaTeX mais ce problème précis n'a jamais été résolu (sur LuaLaTex). Donc, je suis à la case départ!  
https://tex.stackexchange.com/questions/503239/changing-direction-of-critical-footnotes-in-reledmac-with-arabic
Dois-je ouvrir une requette pour examiner ce problème?

**Comment by maieul** [comment]
hum, le mieux serait d'avoir une discussion avec @ralessi, via un nouveau ticket le mentionnant. Cela fait longtemps que je dois écrire des tuto sur la gestion du RTL avec reledmac, mais comme je ne suis moi même pas concerné, je ne suis pas vraiment à l'aise.

mais ma remarque globale est la suivante : lualatex permet de changer de direction sans affecter l'écriture dans les fichiers auxiliaires, contrairement à xelatex. reledmac propose des hooks dans tous les sens pour insérer des commandes à différents niveaux de notes. Donc a priori il y a moyen de trouver. La question est de savoir quels sont les bonnes commandes tant au niveau de reledmac que de lualatex.

**Comment by ralessi** [comment]
En attendant, mon conseil serait d'essayer d'utiliser `lualatex` + `babel` sans `polyglossia` car `babel` est désormais capable de sélectionner automatiquement le sens de l'écriture.

**Comment by maieul** [comment]
oui, babel a fait de grand progrès. Et polyglossia aussi. Du coup moi je
suis perdu entre les deux :)

**Comment by ralessi** [comment]
J'ai vu passer les récentes mises à jour de polyglossia, mais je suis tellement attaché à babel! ;)

**Comment by yadaddi** [comment]
Merci @maieul et @ralessi de vos conseils  et suggestions avec LuaLaTeX et Babel. Décidément, aujourd'hui ce n'est pas mon jour! LuaLaTeX ne marche plus après une mise à jour! Je vais laisser reposer pour un deux jours et je reprendrai là où je suis. Je laisserais un commentaire si jamais le problème initial est résolu.

**Comment by ralessi** [comment]
> Décidément, aujourd'hui ce n'est pas mon jour! LuaLaTeX ne marche plus après une mise à jour! Je vais laisser reposer pour un deux jours et je reprendrai là où je suis.

Pour régler ce problème, je pense qu'il faut passer par une installation complète de texlive: depuis le 2 février dernier, expl3 est passé en version stable et se trouve donc automatiquement chargé dans le noyau LaTeX: cela peut casser une installation incomplète. Comme ce problème est apparu aujourd'hui-même, mieux vaut prendre les devants...

**Comment by yadaddi** [comment]
@ralessi Votre indication sur mon problème de LuaLaTeX est pertinente! Ça marche bien maintenant. Avec LuaLaTeX, ça résout effectivement le problème de \sameword: ça fonctionne très bien. Le problème de l'inversion des numéros de lignes dans les notes critiques reste cependant entier; par exemple: 10-11 en arabe au lieu de 11-10

**Comment by yadaddi** [comment]
> Cela fait longtemps que je dois écrire des tuto sur la gestion du RTL avec reledmac, mais comme je ne suis moi même pas concerné, je ne suis pas vraiment à l'aise.

@maieul Je suis loin de connaître toutes les nuances de Reledmac/Reledpar, mais j'assisterais avec plaisir dans l'élaboration d'un tuto sur la gestion du RTL.

**Comment by ralessi** [comment]
Parfait donc! Pour ce problème de numéros de ligne, je vois que le nombre 10 est imprimé dans l'ordre arabe correct, c'est-à-dire en LTR et non en RTL. Pour 11, évidemment, on ne peut rien dire!

Donc je pense que l'inversion des deux nombres vient de reledmac. Peut-être y a-t-il un `hook` à cet endroit qui permette d'intervenir, @maieul?

**Comment by yadaddi** [comment]
@ralessi Désolé pour la confusion. Il ne s'agit pas des numéros en soi, mais de leur succession après un trait.  Je vais ouvrir un nouveau ticket sur ce point.

**Comment by ralessi** [comment]
> @ralessi Désolé pour la confusion. Il ne s'agit pas des numéros en soi, mais de leur succession après un trait.  Je vais ouvrir un nouveau ticket sur ce point.

Oui, c'est bien ce que j'avais compris: les numéros sont bons, mais chacun devrait être de l'autre côté du tiret. Or comme c'est reledmac qui met le tiret, c'est bien reledmac qu'il faudrait corriger.

**Comment by maieul** [comment]
pour le problème du tiret de séparation
https://tex.stackexchange.com/a/504176/7712

**Comment by yadaddi** [comment]
@maieul Effectivement, ça a été résolu, mais uniquement pour XeLaTeX. La solution ne marche pas pour LuaLaTeX. J'ai fait toutes les combinaisons possibles, mais rien n'y fait!

**Comment by maieul** [comment]
ouvre un ticket spécifique, avec un ECM. 

Normalement LuaLaTeX a tout les outils pour faire facilement des bascules d'un sens à l'autre.

**Comment by yadaddi** [comment]
C'est chose faite! https://github.com/maieul/ledmac/issues/839

**Comment by maieul** [comment]
Je ferme, je vais juste mettre une doc indiquant qu'il faut utiliser LuaTeX pour sameword en RTL.
